src/lib/event_mapper/impl/general_aws_event_mapper.py

```diff
@@ -17,6 +18,15 @@ class GeneralAwsEventMapper(ABC):
         to_notification_messages(dict): maps AWS event object to a list of notification message objects
     """
 
+    def __init__(
+        self,
+        event: dict,
+        settings: Settings
+    ):
+        self.event = event
+        self.settings = settings
```

Comment:
Not sure if we need Settings in the class fields. Looks like we use it only to determine the _monitored_env_name_

Reply:
What do you think about:
1. Assigning Settings only as a **class field**
2. Creating a method which calculates _monitored_env_name_, accepting the event as an argument. _general_aws_event_mapper_ will have the default implementation, while _glue_workflow_event_mapper_ will overrride it. The method will use "_settings_" class field.
3. Accepting only "_event_" as a mapping method argument, and, when _monitored_env_name_ is needed, the mapper will call the corresponding method.

Reply:
Decided to proceed with class fields approach for these classes to be consistent with others like senders, metrics extractors etc.